transactions/stakingCollection/create_machine_account.cdc

```diff
@@ -13,14 +13,21 @@ transaction(nodeID: String,
 
     prepare(account: auth(BorrowValue) &Account) {
         self.stakingCollectionRef = account.storage.borrow<auth(FlowStakingCollection.CollectionOwner) &FlowStakingCollection.StakingCollection>(from: FlowStakingCollection.StakingCollectionStoragePath)
-            ?? panic("Could not borrow a reference to a StakingCollection in the primary user's account")
+            ?? panic(FlowStakingCollection.getCollectionMissingError(nil))
 
         if let machineAccount = self.stakingCollectionRef.createMachineAccountForExistingNode(nodeID: nodeID, payer: account) {
             let sigAlgo = SignatureAlgorithm(rawValue: machineAccountKeySignatureAlgorithm)
-                ?? panic("Could not get a signature algorithm from the raw enum value provided")
+                ?? panic("Cannot create machine account with provided key: Must provide a signature algorithm raw value that corresponds to "
```

Comment:
I think I do want to put this error message construction in a contract somewhere. Any ideas about where it could go? Potentially in the Crypto contract? Or maybe just in the Service Account contract? Another option would be to create a new contract in the service account that just contains multiple different common error message constructions.

Reply:
At first glance, the Crypto contract makes the most sense to me. A version of the enum constructor that panics on invalid input with an error message like this.

Reply:
I agree, Crypto makes the most sense, maybe adding a getter for these enums which reverts with the appropriate message

Reply:
I created an issue for it: https://github.com/onflow/cadence/issues/3617